src/macaron/slsa_analyzer/checks/check_result.py

```diff
@@ -24,6 +24,79 @@ class CheckResultType(str, Enum):
     UNKNOWN = "UNKNOWN"
 
 
+@dataclass
+class Evidence:
+    """The class representing an evidence generated by a check."""
+
+    #: The name of the evidence.
+    name: str
+
+    #: Determines whether the was evidence is found.
+    found: bool
+
+    #: The weight of the evidence.
+    weight: float
+
+
+class EvidenceWeightMap:
```

Comment:
Hmm why do we want to abstract the mapping by creating a class to wrap around `dict[str, Evidence]` here :thinking: ? 

Even though the current implementations is all good, it reminds me of the `Configuration` class https://github.com/oracle/macaron/blob/2d6871fc5d281f71e05e9f3f7ebea6ee84c3bdc1/src/macaron/config/target_config.py#L21 awhile ago, which causes us a lot of trouble. 

Therefore, I think we can go with working with a dictionary directly. We can refactor the `get_max_score` and `get_score` methods into static functions that receives `dict[str, Evidence]` as input parameter.

Reply:
I would say the dict here is an implementation detail, it exists solely to make `update_result` efficient (as it is an index allowing fast lookup of an Evidence by name), so encapsulating it is fine, and allows enforcing the invariant that `key == value.name`. However, this class doesn't fully encapsulate the dict since this detail is exposed to the user by allowing construction with a preexisting dict, so the user can break the invariant anyway. A fully encapsulated design would just take a list of evidence at the constructor and construct the index dict internally based on that.

Reply:
> A fully encapsulated design would just take a list of evidence at the constructor and construct the index dict internally based on that.

I agree on this solution.

Reply:
Yes, the evidence dict did not need to be exposed. See commit [a882a62](https://github.com/oracle/macaron/pull/673/commits/a882a62dda353170d381273e9a1ca7fe235b03e0)